### PR TITLE
Add and remove group members

### PR DESCRIPTION
**Pull request by giarc3**
- Added `groupAddMembers()` to add a list of `UserIds` to a given `GroupId`
- Added `groupRemoveMembers()` to remove a list of `UserIds` to a given `GroupId`
- Added tests that remove the primary user as a group member, then re-add them as a member

### CODE REVIEW

File: src/main/scala/com/ironcorelabs/GroupAccessEditResult.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+case class GroupAccessEditResult(
+  succeeded: List[UserId],
+  failed: List[GroupAccessEditErr]
+)
+
+object GroupAccessEditResult {
+  def apply(gaer: jsdk.GroupAccessEditResult): GroupAccessEditResult =
+    GroupAccessEditResult(
+      gaer.getSucceeded.map(id => UserId(id.getId)).toList,
+      gaer.getFailed.map(err => GroupAccessEditErr(UserId(err.getUser.toString), err.getError)).toList
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/GroupAccessEditResult.scala]: You shouldn't have to `toString` this. Also, I don't think `toString` would have been the right call to make `getId` is. Either way, we have a UserId apply function that takes the java type.

File: src/test/scala/ironoxide/FullIntegrationTest.scala [hunk 2]
```diff
@@ -65,13 +65,58 @@ class FullIntegrationTest extends AsyncWordSpec with Matchers with EitherValues
...
-      groupCreateResult.needsRotation shouldBe Some(false)
+      groupCreateResult.needsRotation.value shouldBe false
+    }
+  }
+
+  "Group Add Members" should {
+    "Fail for nonexistent GroupId" in {
+      val sdk = IronSdkSync[IO](deviceContext)
+      val maybeAddMembersResult = sdk.groupAddMembers(GroupId("kumquat"), Nil).attempt.unsafeRunSync
+      maybeAddMembersResult.isLeft shouldBe true
```
Comment (coltfred) [inline src/test/scala/ironoxide/FullIntegrationTest.scala]: You can also do `maybeAddMembersResult shouldBe 'left`

File: src/test/scala/ironoxide/FullIntegrationTest.scala [hunk 3]
```diff
@@ -65,13 +65,58 @@ class FullIntegrationTest extends AsyncWordSpec with Matchers with EitherValues
...
+    }
+    "Fail for adding an existing member" in {
+      val sdk = IronSdkSync[IO](deviceContext)
+      val addMembersResult = sdk.groupAddMembers(validGroupId, List(primaryTestUserId)).attempt.unsafeRunSync.value
+      addMembersResult.failed.head.error should include("User was already a member")
```
Comment (coltfred) [inline src/test/scala/ironoxide/FullIntegrationTest.scala]: nitpick: I slightly prefer asserting about the length of the list ahead of doing `head`, which will throw.

File: src/main/scala/com/ironcorelabs/GroupAccessEditResult.scala (hunk 1 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/GroupAccessEditResult.scala]: Good catch, I definitely intended to do `.getId`. But I've now removed that and just used the java type constructor.

File: src/test/scala/ironoxide/FullIntegrationTest.scala (hunk 2 above)
Comment (giarc3) [reply src/test/scala/ironoxide/FullIntegrationTest.scala]: I find this way to be more easily understood, and the `'left` syntax will be going away in the future. But let me know if you really want me to change it.

File: src/test/scala/ironoxide/FullIntegrationTest.scala (hunk 3 above)
Comment (giarc3) [reply src/test/scala/ironoxide/FullIntegrationTest.scala]: I added an assertion about length before each call to `head`.

**Pull request merged**